CSharpMath/Structures/Dictionary.cs

```diff
@@ -125,7 +125,7 @@ public class BiDictionary<TFirst, TSecond>
 #pragma warning restore CA1010 // Collections should implement generic interface
 #pragma warning restore CA1710 // Identifiers should have correct suffix
     : ProxyAdder<TFirst, TSecond>
-    where TFirst : IEquatable<TFirst> {
+    where TFirst: class, IEquatable<TFirst> {
```

Comment:
Why this constraint was added?

Reply:
It's so that FirstOrDefault (lookup of any firsts that match svalue) returns an NRT. It is a bit of a halfway house. Could be replaced with an explicit existence check in line 162. Or TFirst could be removed since we are always using BiDictionary with strings. Which do you think is better?

Reply:
I removed the constraint but we could still remove TFirst unless there are reasons to take TFirst other than a string.